client/my-sites/controller.js

```diff
@@ -109,6 +111,50 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
+function isPathAllowedForDomainOnlySite( pathname, site ) {
+	if ( ! get( site, 'options.is_domain_only', false ) ) {
```

Comment:
Should we use the selector (`client/state/selectors/is-domain-only-site.js`) that @gziolo created in #10656?